### PR TITLE
Features for team game, small enhancement for collisions

### PR DESCRIPTION
**Pull request by milosmilosevic13**

### CODE REVIEW

**Review by mmacura9 — APPROVED** [review]

File: source/dagger/gameplay/team_game/team_game_main.cpp [hunk 1]
```diff
@@ -53,3 +56,48 @@ void team_game::SetupWorld(Engine &engine_)
         col.size = sprite.size;
     }
 }
+
+struct Mage {
```
Comment (DPrivalov) [inline source/dagger/gameplay/team_game/team_game_main.cpp]: strange that Mage struct is inside main, and not inside mage.h

File: source/dagger/gameplay/common/simple_collisions.cpp [hunk 2]
```diff
@@ -30,9 +30,11 @@ void SimpleCollisionsSystem::Run()
             {
                 collision.colided = true;
                 collision.colidedWith = *it2;
+                collision.colisions.emplace_back(*it2);
```
Comment (DPrivalov) [inline source/dagger/gameplay/common/simple_collisions.cpp:43]: don't forget to clean this list at the beginning of this function. You can reset colidedWith as well.

just add one more loop before 'while(it != view.end())'

**Review by DPrivalov — APPROVED** [review]
simple collision changes should be a separate pull request, but it's ok this time.

File: source/dagger/gameplay/team_game/team_game_main.cpp (hunk 1 above)
Comment (milosmilosevic13) [reply source/dagger/gameplay/team_game/team_game_main.cpp]: It was there initially just for some testing, it'll be moved.

**Comment by milosmilosevic13** [comment]
Needed to test mage attacking and range of attack, so I had to add some things that should have been a separate features/pull requests. Last commit of this pull.

**Review by nkristina — APPROVED** [review]

**Pull request merged**